camelot/view/action_steps/application.py

```diff
@@ -77,20 +77,24 @@ class MainWindow(ActionStep):
 
     """
 
-    admin: ApplicationAdmin
+    admin: InitVar[ApplicationAdmin]
     window_title: str = field(init=False)
 
-    def __post_init__(self):
-        self.window_title = self.admin.get_name()
+    admin_route: Route = field(init=False)
 
-    def render(self, gui_context):
+    def __post_init__(self, admin):
+        self.window_title = admin.get_name()
+        self.admin_route = admin.get_admin_route()
+
+    @classmethod
+    def render(cls, gui_context, step):
         """create the main window. this method is used to unit test
         the action step."""
         from ..mainwindowproxy import MainWindowProxy
 
         main_window_context = gui_context.copy()
         main_window_context.progress_dialog = None
-        main_window_context.admin = self.admin
+        main_window_context.admin = AdminRoute.admin_for(tuple(step["admin_route"]))
```

Comment:
het zetten van de main_window_context.admin zou in principe niet meer nodig moeten zijn,
enkel de main_window_context.admin_route zetten zou nodig moeten zijn.

kan je aub uittesten of dit in de praktijk ook het geval is ?